habitat/utils/profiling_utils.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+r"""Helper functions for profiling. Profilers like Nvidia Nsight can capture the
+time spent in annotated ranges.
+
+Example of range_push/range_pop usage:
+def my_function():
+    range_push("my_function")
+    for _ in range(1000)
+        range_push("my loop iteration")
+        # do stuff
+        range_pop()  # my loop iteration
+    range_pop()  # my_function
+
+Example of capturing an Nsight Systems profile:
+apt-get install nvidia-nsight
+export HABITAT_PROFILING=1
+path/to/nvidia/nsight-systems/bin/nsys profile --sample=none --trace=cuda,nvtx
+--trace-fork-before-exec=true --output=my_profile python my_program.py
+# look for my_profile.qdrep in working directory
+"""
+import os
+
+env_var = os.environ.get("HABITAT_PROFILING", "0")
```

Comment:
For a another example of using environment variables in Hab-Lab, look at `SLURM_JOB_ID` in `ddp_utils.py`.